Add Adapt Map (#7640)
sunpy/sunpy: docs/whatsnew/6.0.rst

Support for ADvanced Adaptive Prediction Technique (ADAPT)
==========================================================

A new map source has been added (`sunpy.map.sources.ADAPTMap`) to support the ADAPT data files.
In addition, a new client (`sunpy.net.dataretriever.ADAPTClient`) has been added to search and download ADAPT files.

> Please note that there are several data arrays within the FITS file and map will try to read them all and fail, so you have to specify the Header Data Pair you want to read.
> In this case, it will always be the first one. You can do this by passing the ``hdus`` argument to the `sunpy.map.Map` call.

Reviewer: ```suggestion
Please note that there are several data arrays within the ADAPT FITS files and `~sunpy.map.Map` will, by default, try to read them all and fail.
In these cases, you must specify the header-data Pair you want to read.
For these data, it will always be the first one.
You can specify this by passing the ``hdus`` keyword argument to `~sunpy.map.Map`,
```
